firestore/collgroupref.go

```diff
@@ -84,20 +84,21 @@ func (c *CollectionGroupRef) PartitionQuery(ctx context.Context, count int) (Que
 			break
 		}
 		if err != nil {
-			return q, nil, fmt.Errorf("PartitionQuery: %v", err)
+			return &q, nil, fmt.Errorf("PartitionQuery: %v", err)
 		}
-		doc := cursor.Values[0].GetReferenceValue()
-		docs = append(docs, doc)
+		docRef := cursor.Values[0].GetReferenceValue()
+		// Splitting values to pass it to "fieldValuesToCursorValues",
+		// since cursor values are not yet supported,
+		// and "Value_ReferenceValue" is taking q.path + " / " + docID.
+		doc := strings.Split(docRef, c.path+"/")
+		docs = append(docs, doc[1])
 	}
-	for i, doc := range docs {
-		partition := q
-		if i == 0 {
-			partitions = append(partitions, partition.EndBefore(doc))
-		} else if i == len(docs)-1 {
-			partitions = append(partitions, partition.StartAt(doc))
-		} else {
-			partitions = append(partitions, partition.StartAt(doc).EndBefore(docs[i+1]))
+	if docs != nil {
+		partitions = append(partitions, q.EndBefore(docs[0]))
+		for i := 0; i < len(docs)-1; i++ {
+			partitions = append(partitions, q.StartAt(docs[i]).EndBefore(docs[i+1]))
```

Comment:
Aren't you creating a partition for every doc now?

Reply:
You're walking through every doc and adding a partition for every doc. It looks like you'll get partition for every doc instead of three partitions.

Reply:
No, I'll try to explain to you more clear.
1. We have 3 cursors from `it := client.PartitionQuery(ctx, req)`. A cursor is the position from which we start reading documents or it is the position at which we finish reading documents. 
2. After that we take the document path from the each cursor. (So we have 3 documents)
3. And we splitting.
4. For first partition we do query with startAt from the beginning and endBefore first cursor document` partitions = append(partitions, q.EndBefore(docs[0]))`
5. For last partition we set query with starting from last cursor to end of documents, 'cause our documents were sorted before Asc.
6. And other two partitions will start from 0 cursor doc path to 1st and 1st to 2nd.

Reply:
[Java PR partition example](https://github.com/googleapis/java-firestore/pull/202/files#diff-a5ee80f101c23a4cd99d2db070c9ff61f0d96a3d1a948d7c080278d9c54de450R570-R574)